src/fidesops/models/privacy_request.py

```diff
@@ -226,6 +228,77 @@ def get_results(self) -> Dict[str, Any]:
         result_prefix = f"{self.id}__*"
         return cache.get_encoded_objects_by_prefix(result_prefix)
 
+    def cache_paused_step_and_collection(
+        self,
+        paused_step: Optional[ActionType] = None,
+        paused_collection: Optional[CollectionAddress] = None,
+    ) -> None:
+        """
+        When a privacy request is paused, cache both the paused step (access or erasure) and the collection
+        awaiting manual input.  For example, we might pause a privacy request at the erasure step of the
+        postgres_example:address collection.
+
+        The paused_step is needed because we may build and execute multiple graphs as part of running a privacy request.
+        An erasure request builds two graphs, one to access the data, and the second to mask it.
+        We need to know if we should resume execution from the access or the erasure portion of the request.
+        """
+        cache: FidesopsRedis = get_cache()
+        paused_key = f"PAUSED_LOCATION__{self.id}"
+
+        cache.set_encoded_object(
+            paused_key,
+            f"{paused_step.value}__{paused_collection.value}"
+            if paused_step and paused_collection
+            else None,
+        )
+
+    def get_paused_step_and_collection(
+        self,
+    ) -> Tuple[Optional[ActionType], Optional[CollectionAddress]]:
+        """Get both the paused step (access or erasure) and collection awaiting manual input for the given privacy request.
+
+        The paused step lets us know if we should resume privacy request execution from the "access" or the "erasure"
+        portion of the privacy request flow, and the collection tells us where we should cache manual input data for later use,
+        In other words, this manual data belongs to this collection.
+        """
+        cache: FidesopsRedis = get_cache()
+        cached_paused_node = cache.get_encoded_objects_by_prefix(
+            f"PAUSED_LOCATION__{self.id}"
+        )
+        node_addr = (
+            list(cached_paused_node.values())[0]
+            if len(cached_paused_node.keys()) == 1
+            else None
+        )
+
+        if node_addr:
+            split_addr = node_addr.split("__")
+            return ActionType(split_addr[0]), CollectionAddress.from_string(
+                split_addr[1]
```

Comment:
Shouldn't be, `get_paused_step_and_collection` is retrieving the data from the method above: `cache_paused_step_and_collection`, which stores two values separated by a "__".  Maybe I should make the separator even more unique in case the CollectionAddress happened to have two underscores. I'll dry up the separator and share between the two methods too.